iml-manager-cli/src/stratagem.rs

```diff
@@ -0,0 +1,341 @@
+        return Err(DurationParseError::InvalidValue.into());
+    }
+
+    let mut val = String::from(src);
+    let unit = val.pop();
+
+    let val = val.parse::<u64>()?;
+
+    match unit {
+        Some('h') => Ok(val * 3_600_000),
+        Some('d') => Ok(val * 86_400_000),
+        Some('m') => Ok(val * 60_000),
+        Some('s') => Ok(val * 1_000),
+        Some('1'...'9') => Err(DurationParseError::NoUnit.into()),
+        _ => Err(DurationParseError::InvalidUnit.into()),
+    }
+}
+
+fn handle_cmd_resp(
+    (resp, body): (iml_manager_client::Response, iml_manager_client::Chunk),
+) -> Result<CmdWrapper, ImlManagerCliError> {
+    let status = resp.status();
+    if status.is_success() {
+        Ok(serde_json::from_slice::<CmdWrapper>(&body)?)
+    } else if status.is_client_error() {
+        log::debug!("body: {:?}", std::str::from_utf8(&body));
+        serde_json::from_slice::<RunStratagemValidationError>(&body)
+            .map(std::convert::Into::into)
+            .map(Err)?
+    } else if status.is_server_error() {
+        Err(RunStratagemValidationError {
+            code: RunStratagemCommandResult::ServerError,
+            message: "Internal server error.".into(),
+        }
+        .into())
+    } else {
+        Err(RunStratagemValidationError {
+            code: RunStratagemCommandResult::UnknownError,
+            message: "Unknown error.".into(),
+        }
+        .into())
+    }
+}
+
+pub fn stratagem_cli(command: StratagemCommand) {
+    match command {
+        StratagemCommand::Scan(data) => {
+            let fut = post("run_stratagem", data).and_then(handle_cmd_resp);
+
+            let command: Result<CmdWrapper, ImlManagerCliError> = run_cmd(fut);
+
+            match command {
+                Ok(CmdWrapper { command }) => {
+                    let stop_spinner = start_spinner(&command.message);
+
+                    let command = run_cmd(wait_for_cmd(command)).expect("Could not poll command");
+
+                    stop_spinner(None);
+
+                    display_cmd_state(&command);
+
+                    exit(exitcode::OK)
+                }
+                Err(validation_error) => {
+                    display_error(validation_error);
+
+                    exit(exitcode::CANTCREAT)
+                }
+            }
+        }
+        StratagemCommand::StratagemConfig(x) => match x {
+            StratagemConfig::List => {
+                let stop_spinner = start_spinner("Finding config...");
+
+                let fut = get(
+                    StratagemConfiguration::endpoint_name(),
+                    serde_json::json!({ "limit": 0 }),
+                );
+                let result: Result<ApiList<StratagemConfiguration>, _> = run_cmd(fut);
+
+                stop_spinner(None);
+
+                match result {
+                    Ok(r) => {
+                        if r.objects.is_empty() {
+                            return println!("0 Stratagem configuration records found");
```

Comment:
Any reason why we don't call `exit(exitcode::OK)`?

Reply:
No difference really:

```
[root@adm chroma-manager]# iml stratagem configure list
0 Stratagem configuration records found
[root@adm chroma-manager]# echo $?
0
```